### PR TITLE
Clean up help link

### PR DESCRIPTION
**Pull request by tmat**
Removes ProjectType parameter from Bing help link query.
Adds `Language` property to `DiagnosticData` so that it doesn't need to be calculated from `ProjectId` and `Workspace` later.

Fixes perf issue tracked by: https://github.com/dotnet/roslyn/issues/26717
Fixes: https://github.com/dotnet/roslyn/issues/37722

Recommended to review each commit separately.

### CODE REVIEW

**Comment by tmat** [comment]
@jasonmalinowski @sharwell PTAL

**Comment by tmat** [comment]
@jasonmalinowski @sharwell Ping

File: src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/BaseDiagnosticItem.cs [hunk 1]
```diff
@@ -8,86 +8,47 @@
...
+        public DiagnosticDescriptor Descriptor { get; }
+        public ReportDiagnostic EffectiveSeverity { get; private set; }
 
         public override event PropertyChangedEventHandler PropertyChanged;
 
-
-        public BaseDiagnosticItem(DiagnosticDescriptor descriptor, ReportDiagnostic effectiveSeverity)
-            : base(string.Format("{0}: {1}", descriptor.Id, descriptor.Title))
+        public BaseDiagnosticItem(DiagnosticDescriptor descriptor, ReportDiagnostic effectiveSeverity, string language)
+            : base(descriptor.Id + ": " + descriptor.Title)
```
Comment (sharwell) [inline src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/BaseDiagnosticItem.cs]:
```suggestion
            : base($"{descriptor.Id}: {descriptor.Title}")
```

File: src/VisualStudio/Core/Def/Implementation/Utilities/BrowserHelper.cs [hunk 2]
```diff
@@ -35,29 +37,67 @@ public static bool TryGetUri(string link, out Uri uri)
...
+            => GetHelpLink(data.Id, data.ENUMessageForBingSearch, data.Language, data.HelpLink);
+
+        private static Uri GetHelpLink(string diagnosticId, string title, string language, string rawHelpLink)
         {
-            var errorCode = diagnostic.Id;
-            var title = diagnostic.ENUMessageForBingSearch;
-            workspace.GetLanguageAndProjectType(diagnostic.ProjectId, out var language, out var projectType);
+            if (string.IsNullOrWhiteSpace(diagnosticId))
+            {
+                return default;
```
Comment (sharwell) [inline src/VisualStudio/Core/Def/Implementation/Utilities/BrowserHelper.cs]: `default` makes it look like the return type is a struct.

```suggestion
                return null;
```

File: src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/BaseDiagnosticItem.cs (hunk 1 above)
Comment (tmat) [reply src/VisualStudio/Core/SolutionExplorerShim/DiagnosticItem/BaseDiagnosticItem.cs]: Interpolated string is less efficient and doesn't make it more readable.

**Pull request merged**